src/pages/flood-control-projects/map.tsx

```diff
@@ -250,42 +330,128 @@ const FloodControlProjectsMap: React.FC = () => {
     }
   }, [filteredProjects, selectedRegion]);
 
-  // Handle region click
+  // Debounced region click handler
   const onRegionClick = useCallback(
-    (feature: GeoJSON.Feature<GeoJSON.Geometry, RegionProperties>) => {
-      if (!feature.properties) return;
+    async (feature: GeoJSON.Feature<GeoJSON.Geometry, RegionProperties>) => {
+      console.log('Region click handler called');
+      if (!feature.properties || isRegionClicking) return;
+
       const props = feature.properties;
       const regionName = props.name;
 
-      // Get the bounding box of the region and calculate geo search parameters
-      const bounds = L.geoJSON(feature.geometry).getBounds();
-      const geoParams = calculateGeoSearchParams(bounds);
-      setGeoSearch(geoParams);
-
-      // Set loading state first
-      const regionDetails: RegionData = {
-        id: regionName,
-        name: regionName,
-        loading: true,
-      };
-      setSelectedRegion(regionDetails);
-
-      // Only zoom/fit bounds if we're not already zoomed in (zoom level <= 8)
-      if (mapRef.current && feature.geometry && zoomLevel <= 8) {
-        mapRef.current.fitBounds(bounds, { padding: [20, 20] });
-        // Force zoom to at least level 9 to show project pins
-        setTimeout(() => {
-          if (mapRef.current && mapRef.current.getZoom() < 9) {
-            mapRef.current.setZoom(9);
-          }
-          setZoomLevel(mapRef.current?.getZoom() || 9);
-        }, 500);
-      } else if (mapRef.current) {
-        // If already zoomed in, just update the zoom level state without changing the view
-        setZoomLevel(mapRef.current.getZoom());
+      // Clear any existing timeout
+      if (clickTimeoutRef.current) {
+        clearTimeout(clickTimeoutRef.current);
```

Comment:
I think we can do this cleaner by pulling it out, clearing it then updating the ref iirc

Reply:
done